src/test/ui/consts/recursive-zst-static.default.stderr

```diff
@@ -1,8 +1,8 @@
 error[E0391]: cycle detected when const-evaluating `FOO`
-  --> $DIR/recursive-zst-static.rs:10:18
+  --> $DIR/recursive-zst-static.rs:10:1
    |
 LL | static FOO: () = FOO;
-   |                  ^^^
+   | ^^^^^^^^^^^^^^^^^^^^^
```

Comment:
This is due to the fact that you're using the `root_span` now. The invocation of the `const_eval` query should be using a more precise span

Reply:
So that would be [here](https://github.com/rust-lang/rust/pull/72879/files#diff-470c35c1f39f9d3f8817435a63c73ad2L846) or [here](https://github.com/rust-lang/rust/pull/72879/files#diff-470c35c1f39f9d3f8817435a63c73ad2L877)? But I am taking into account the current `Location` there.

Reply:
Ah but it's a static so this is going through memory which cannot access the `Location`. But then it should be using `DUMMY_SP`...

Reply:
uhmm... `InterpCx::const_eval_raw` is dead code except for `try_eval_const_fn_call` and some use in miri that could be replaced by using `const_eval` instead?

Reply:
I'm guessing these sites are actually from the `const_eval_raw` invocation from `Memory`

Reply:
> uhmm... InterpCx::const_eval_raw is dead code except for try_eval_const_fn_call and some use in miri that could be replaced by using const_eval instead?

Someone argued we should use "raw" instead... I thought it was you.^^
Anyway with https://github.com/rust-lang/rust/issues/72396 this will all solve itself.

> I'm guessing these sites are actually from the const_eval_raw invocation from Memory

That's my guess, too. But then the span is `DUMMY_SP`, so I wonder how it figures out the span that it ends up using.

Reply:
Threading the `Span` through all the memory access methods is quite the mess, though... the separation of `Memory` from the main eval context (or at least, having all these methods on the memory and not the context) more and more often looks like a mistake.^^

Reply:
It seemed like a nice idea for modularity reasons at the time. But just keeping the inherent methods in their own module should totally suffice.

The diff for doing such a change should be fairly easy to review, too, as all the major changes will be replacing `x.memory` with `x`

Reply:
Yeah.

The question is now, is it okay to temporarily regress cycle error spans (so we land this PR first and do the memory thing eventually -- I have some plans for changes to the `Allocation` API anyway), or should we leave `TyCtxtAt` in `Memory` for now to avoid span regressions?

Reply:
I'm fine regressing the spans for now, it's not too bad. Please open an issue about it though

Reply:
Done: https://github.com/rust-lang/rust/issues/73283